timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -96,15 +96,20 @@ def _constructor(self) -> Type[TimeSeriesDataFrame]:
 
     @property
     def freq(self):
-        ts_index = self.index.levels[1]  # noqa
+        ts_index = self.index.get_level_values(1)  # noqa
+        first_item = self.index.get_level_values(0)[0]
         freq = (
             ts_index.freq
             or ts_index.inferred_freq
-            or self.loc[0].index.freq  # fall back to freq of first item
-            or self.loc[0].index.inferred_freq
+            or self.loc[first_item].index.freq  # fall back to freq of first item
+            or self.loc[first_item].index.inferred_freq
```

Comment:
Wouldn't it make more sense to look at `freq` / `inferred_freq` of each individual time series to ensure that 
1. The frequencies are consistent across series in the dataset
2. It doesn't happen that other time series (not the first one) have an irregular index?

Something like the following snippet
```python
def get_freq(series):
    return series.index.freq or series.index.inferred_freq

freq_for_series = [get_freq(self.loc[idx]) for idx in self.iter_items()]
if len(set(freq_for_series)) > 1 or freq_for_series[0] is None:
    raise ValueError("...")
```

Reply:
If it is cheap to do, this will be valuable and I agree it makes sense to check per time series.

Reply:
It's actually not very cheap. For one of the data sets I'm looking at, this takes about ~5sec for 10K items with each about 30 time steps. Do we want to pay this or maybe just look at top 100 items as a middle ground solution?

Reply:
also note this will be called every time freq is accessed. so we might want to cache this value as well.